action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -42,24 +43,27 @@ type (
 
 	// StateDBAdapter represents the state db adapter for evm to access iotx blockchain
 	StateDBAdapter struct {
-		sm                         protocol.StateManager
-		logs                       []*action.Log
-		transactionLogs            []*action.TransactionLog
-		err                        error
-		blockHeight                uint64
-		executionHash              hash.Hash256
-		lastAddBalanceAddr         string
-		lastAddBalanceAmount       *big.Int
-		refund                     uint64
-		refundSnapshot             map[int]uint64
-		cachedContract             contractMap
-		contractSnapshot           map[int]contractMap   // snapshots of contracts
-		suicided                   deleteAccount         // account/contract calling Suicide
-		suicideSnapshot            map[int]deleteAccount // snapshots of suicide accounts
-		preimages                  preimageMap
-		preimageSnapshot           map[int]preimageMap
-		accessList                 *accessList // per-transaction access list
-		accessListSnapshot         map[int]*accessList
+		sm                   protocol.StateManager
+		logs                 []*action.Log
+		transactionLogs      []*action.TransactionLog
+		err                  error
+		blockHeight          uint64
+		executionHash        hash.Hash256
+		lastAddBalanceAddr   string
+		lastAddBalanceAmount *uint256.Int
+		refund               uint64
+		refundSnapshot       map[int]uint64
+		cachedContract       contractMap
+		contractSnapshot     map[int]contractMap   // snapshots of contracts
+		suicided             deleteAccount         // account/contract calling Suicide
+		suicideSnapshot      map[int]deleteAccount // snapshots of suicide accounts
+		preimages            preimageMap
+		preimageSnapshot     map[int]preimageMap
+		accessList           *accessList // per-transaction access list
+		accessListSnapshot   map[int]*accessList
+		// Transient storage
+		transientStorage           transientStorage
+		transientStorageSnapshot   map[int]transientStorage
```

Comment:
this PR has lots of change, and focus on making interface work with Cancun.
let's enable transient storage in another PR